pkg/discovery/monitoring/master/cluster_monitor.go

```diff
@@ -0,0 +1,301 @@
+		metrics.Memory, metrics.Capacity, memoryCapacityKiloBytes)
+	nodeResourceMetrics = append(nodeResourceMetrics, nodeCpuCapacityCoreMetrics, nodeMemoryCapacityKiloBytesMetrics)
+
+	// The provisioned resources have the same capacities.
+	// cpuProvisioned and memoryProvisioned
+	cpuProvisionedCapacityCore := cpuCapacityCore
+	memoryProvisionedCapacityKiloBytes := memoryCapacityKiloBytes
+	glog.V(4).Infof("Cpu provisioned capacity of node %s is %f core", node.Name, cpuProvisionedCapacityCore)
+	glog.V(4).Infof("Memory provisioned capacity of node %s is %f Kb", node.Name, memoryProvisionedCapacityKiloBytes)
+	nodeCpuProvisionedCapacityCoreMetrics := metrics.NewEntityResourceMetric(task.NodeType, key,
+		metrics.CPUProvisioned, metrics.Capacity, cpuProvisionedCapacityCore)
+	nodeMemoryProvisionedCapacityKiloBytesMetrics := metrics.NewEntityResourceMetric(task.NodeType, key,
+		metrics.MemoryProvisioned, metrics.Capacity, memoryProvisionedCapacityKiloBytes)
+	nodeResourceMetrics = append(nodeResourceMetrics, nodeCpuProvisionedCapacityCoreMetrics, nodeMemoryProvisionedCapacityKiloBytesMetrics)
+
+	// Provisioned resource used is the sum of all the requested resource of all the pods running in the node.
+	nodeCpuProvisionedUsedCore, nodeMemoryProvisionedUsedKiloBytes, err := util.GetNodeResourceRequestConsumption(m.nodePodList[node.Name])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get provision used metrics: %s", err)
+	}
+	glog.V(4).Infof("Cpu provisioned used of node %s is %f core", node.Name, nodeCpuProvisionedUsedCore)
+	glog.V(4).Infof("Memory provisioned used of node %s is %f Kb", node.Name, nodeMemoryProvisionedUsedKiloBytes)
+	nodeCpuProvisionedUsedCoreMetrics := metrics.NewEntityResourceMetric(task.NodeType, key,
+		metrics.CPUProvisioned, metrics.Used, nodeCpuProvisionedUsedCore)
+	nodeMemoryProvisionedUsedKiloBytesMetrics := metrics.NewEntityResourceMetric(task.NodeType, key,
+		metrics.MemoryProvisioned, metrics.Used, nodeMemoryProvisionedUsedKiloBytes)
+	nodeResourceMetrics = append(nodeResourceMetrics, nodeCpuProvisionedUsedCoreMetrics, nodeMemoryProvisionedUsedKiloBytesMetrics)
+
+	return nodeResourceMetrics, nil
+}
+
+// Parse the labels of a node and create one EntityStateMetric
+func parseNodeLabels(node *api.Node) metrics.EntityStateMetric {
+	labelsMap := node.ObjectMeta.Labels
+	if len(labelsMap) > 0 {
+		var labels []string
+		for key, value := range labelsMap {
+			l := key + "=" + value
+			glog.V(4).Infof("label for this Node is : %s", l)
+
+			labels = append(labels, l)
+		}
+		return metrics.NewEntityStateMetric(task.NodeType, util.NodeKeyFunc(node), metrics.Access, labels)
+	} else {
+		return metrics.EntityStateMetric{}
+	}
+}
+
+// ----------------------------------------------- Pod State -------------------------------------------------
+func (m *ClusterMonitor) findPodStates() {
+	for _, podList := range m.nodePodList {
+		for _, p := range podList.Items {
+			pod := p
```

Comment:
I was wrong. It works fine in this case. I was worried it would break at other place where it builds slice like `[]*api.Pod` and when it is used in sth like
```go
for _, p := range pods {
  go func(pod *api.Pod) {...} (&p)
}
```

I am not sure if this java style `for-loop` outperforms `for...range...`  in go